tests/test_grpc.py

```diff
@@ -399,7 +397,10 @@ def test_end_training_round_duplicated_updates(  # pylint: disable=unused-argume
     # participant can only send updates once in a single round
     participant_stub.Rendezvous(RendezvousRequest())
 
-    participant_stub.EndTrainingRound(EndTrainingRoundRequest())
+    participant_store.write_weights("participant1", 0, np.ndarray([]))
+    participant_stub.EndTrainingRound(
+        EndTrainingRoundRequest(participant_id="participant1")
+    )
```

Comment:
any reason you didn't use the `end_training_round` fixture like in `test_coordinator.py`?
similar with some other tests in this file.

Reply:
You means the `end_training_request` fixture right? I don't use it in `test_grpc.py` because it updates the coordinator directly, whereas in this module we want to test the IO layer by sending messages through the "participant stubs".